internal/resource/config/datastore/jdbc_data_store.go

```diff
@@ -0,0 +1,386 @@
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+	client "github.com/pingidentity/pingfederate-go-client/v1125/configurationapi"
+	"github.com/pingidentity/terraform-provider-pingfederate/internal/acctest/common/pointers"
+	internaljson "github.com/pingidentity/terraform-provider-pingfederate/internal/json"
+	"github.com/pingidentity/terraform-provider-pingfederate/internal/resource/config"
+	internaltypes "github.com/pingidentity/terraform-provider-pingfederate/internal/types"
+)
+
+var (
+	jdbcTagConfigAttrType = types.ObjectType{
+		AttrTypes: map[string]attr.Type{
+			"connection_url": basetypes.StringType{},
+			"tags":           basetypes.StringType{},
+			"default_source": basetypes.BoolType{},
+		},
+	}
+
+	jdbcDataStoreAttrType = map[string]attr.Type{
+		"max_pool_size":                basetypes.Int64Type{},
+		"connection_url_tags":          basetypes.SetType{ElemType: jdbcTagConfigAttrType},
+		"type":                         basetypes.StringType{},
+		"name":                         basetypes.StringType{},
+		"blocking_timeout":             basetypes.Int64Type{},
+		"idle_timeout":                 basetypes.Int64Type{},
+		"min_pool_size":                basetypes.Int64Type{},
+		"driver_class":                 basetypes.StringType{},
+		"connection_url":               basetypes.StringType{},
+		"user_name":                    basetypes.StringType{},
+		"allow_multi_value_attributes": basetypes.BoolType{},
+		"validate_connection_sql":      basetypes.StringType{},
+		"password":                     basetypes.StringType{},
+	}
+
+	jdbcDataStoreEmptyStateObj = types.ObjectNull(jdbcDataStoreAttrType)
+)
+
+func toSchemaJdbcDataStore() schema.SingleNestedAttribute {
+	jdbcDataStoreSchema := schema.SingleNestedAttribute{}
+	jdbcDataStoreSchema.Description = "A JDBC data store."
+	jdbcDataStoreSchema.Default = objectdefault.StaticValue(types.ObjectNull(jdbcDataStoreAttrType))
+	jdbcDataStoreSchema.Computed = true
+	jdbcDataStoreSchema.Optional = true
+	jdbcDataStoreSchema.Attributes = map[string]schema.Attribute{
+		"type": schema.StringAttribute{
+			Description: "The data store type.",
+			Computed:    true,
+			Optional:    false,
+			Default:     stringdefault.StaticString("JDBC"),
+		},
+		"password": schema.StringAttribute{
+			Description: "The password needed to access the database. GETs will not return this attribute. To update this field, specify the new value in this attribute.",
+			Optional:    true,
+			Sensitive:   true,
+		},
+		"name": schema.StringAttribute{
+			Description: "The data store name with a unique value across all data sources. Omitting this attribute will set the value to a combination of the connection url and the username.",
+			Computed:    true,
+			Optional:    true,
```

Comment:
Add defaults for these computed and optional fields. And if you can add defaults, remove UseStateForUnknown.

Reply:
Added a ModifyPlan for the names where necessary